[chore] Refactor Coralogix exporter

iblancasa opened the pull request:
#### Description

Some refactoring for the Coralogix exporter:
- Unify signal logic in a single place
- Moved some helpers to `retry` file
- Added unit tests

douglascamata commented on exporter/coralogixexporter/helpers.go:
These all look like they belong to some sort of retry logic. I think we could rename this to `retry.go`. I'm not a big fan of files/packages with super generic names.

douglascamata commented on exporter/coralogixexporter/helpers.go:
```diff
@@ -0,0 +1,90 @@
+// common OTLP logic around request handling such as retries and throttling.
+func processError(err error) error {
+	if err == nil {
+		// Request is successful, we are done.
```
Pretty obvious and unnecessary comment.

douglascamata commented on exporter/coralogixexporter/helpers.go:
```diff
@@ -0,0 +1,90 @@
+		return nil
+	}
+
+	// We have an error, check gRPC status code.
```
Pretty obvious and unnecessary comment.

douglascamata commented on exporter/coralogixexporter/helpers.go:
```diff
@@ -0,0 +1,90 @@
+
+	st := status.Convert(err)
+	if st.Code() == codes.OK {
+		// Not really an error, still success.
```
Pretty obvious and unnecessary comment.

douglascamata commented on exporter/coralogixexporter/helpers.go:
```diff
@@ -0,0 +1,90 @@
+	retryInfo := getRetryInfo(st)
+
+	if !shouldRetry(st.Code(), retryInfo) {
+		// It is not a retryable error, we should not retry.
```
Pretty obvious and unnecessary comment.

iblancasa commented on exporter/config.yaml:
```diff
@@ -0,0 +1,36 @@
+receivers:
```
Remove test file

iblancasa committed:
Refactor Coralogix exporter:
- Unify signal exporter logix in a common place
- Add unit tests
- Move some helper functions to retry file

Signed-off-by: Israel Blancas <iblancasa@gmail.com>
